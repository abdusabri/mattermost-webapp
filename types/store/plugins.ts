// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {ClientPluginManifest} from 'mattermost-redux/types/plugins';
import {Post} from 'mattermost-redux/src/types/posts';
import {IDMappedObjects} from 'mattermost-redux/types/utilities';

export type PluginsState = {
    plugins: IDMappedObjects<ClientPluginManifest>;

    components: {
        [componentName: string]: PluginComponent[];
    };

    postTypes: {
        [postType: string]: PostPluginComponent;
    };
    postCardTypes: {
        [postType: string]: PostPluginComponent;
    };

    adminConsoleReducers: {
        [pluginId: string]: any;
    };
    adminConsoleCustomComponents: {
        [pluginId: string]: AdminConsolePluginComponent;
    };
};

export type PluginComponent = {
    id: string;
    pluginId: string;
    component: React.Component;
};

export type PostPluginComponent = {
    id: string;
    pluginId: string;
    type: string;
    component: React.ComponentType<{post: Post}>;
};

export type AdminConsolePluginComponent = {
    pluginId: string;
    key: string;
    component: React.Component;
    options: {
        showTitle: boolean;
    };
};
