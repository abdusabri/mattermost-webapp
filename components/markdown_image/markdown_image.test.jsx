// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import store from 'stores/redux_store.jsx';
import SingleImageView from 'components/single_image_view';

import MarkdownImage from './markdown_image';

describe('components/MarkdownImage', () => {
    const baseProps = {
        imageMetadata: {
            format: 'png',
            height: 165,
            width: 1041,
        },
        alt: 'test image',
        className: 'markdown-inline-img',
        isEmbedVisible: true,
        postId: 'post_id',
    };

    test('should match snapshot', () => {
        const props = {...baseProps, src: '/images/logo.png'};
        const wrapper = shallow(
            <MarkdownImage {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render a link if the source is unsafe', () => {
        const props = {...baseProps, src: '/images/logo.png'};
        const wrapper = shallow(
            <MarkdownImage {...props}/>
        );
        const childrenWrapper = shallow(wrapper.dive({context: {store}}).props().children());

        expect(childrenWrapper.find('a.markdown__link')).toHaveLength(1);
        expect(childrenWrapper).toMatchSnapshot();
    });

    test('should render a SingleImageView if the source is safe', () => {
        const props = {...baseProps, src: 'https://example.com/image.png'};
        const wrapper = shallow(
            <MarkdownImage {...props}/>
        );
        const childrenNode = wrapper.dive({context: {store}}).props().children(props.src);

        expect(childrenNode.type.displayName).toEqual(SingleImageView.displayName);
        expect(childrenNode).toMatchSnapshot();
    });
});
