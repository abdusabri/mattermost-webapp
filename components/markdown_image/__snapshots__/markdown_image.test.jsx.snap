// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MarkdownImage should match snapshot 1`] = `
<Connect(ExternalImage)
  imageMetadata={
    Object {
      "format": "png",
      "height": 165,
      "width": 1041,
    }
  }
  src="/images/logo.png"
>
  <Component />
</Connect(ExternalImage)>
`;

exports[`components/MarkdownImage should render a SingleImageView if the source is safe 1`] = `
<Connect(SingleImageView)
  fileInfo={
    Object {
      "extension": "png",
      "height": 165,
      "link": "https://example.com/image.png",
      "name": "test image",
      "width": 1041,
    }
  }
  isEmbedVisible={true}
  postId="post_id"
/>
`;

exports[`components/MarkdownImage should render a link if the source is unsafe 1`] = `
<a
  className="theme markdown__link"
  href="/images/logo.png"
  rel="noopener noreferrer"
  target="_blank"
>
  test image
</a>
`;
